test/uk/gov/hmrc/securemessage/connectors/ChannelPreferencesConnectorSpec.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.connectors
+
+import org.mockito.Mockito.when
+import org.mockito.ArgumentMatchers._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.Configuration
+import uk.gov.hmrc.http.{ HeaderCarrier, HttpClient, HttpResponse }
+
+import play.api.http.Status.{ BAD_GATEWAY, BAD_REQUEST, OK }
+import uk.gov.hmrc.emailaddress.EmailAddress
+import uk.gov.hmrc.securemessage.models.core.Identifier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, Future }
+
+@SuppressWarnings(Array("org.wartremover.warts.All"))
+class ChannelPreferencesConnectorSpec extends PlaySpec with ScalaFutures with MockitoSugar {
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  private val validEmailVerification = """{"address":"some@email.com","timestamp":"1987-03-20T01:02:03.000Z"}"""
+  private val inValidEmailVerification = """{"add":"some@email.com","timestamp":"1987-03-20T01:02:03.000Z"}"""
+  private val inValidJson = """{"add:"some@email.com","timestamp":"1987-03-20T01:02:03.000Z"}"""
+  val cpURL =
+    "https://host:443/channel-preferences/preference/email?enrolmentKey=HMRC-CUS-ORG&taxIdName=EORINumber&taxIdValue=eoriid"
+
+  "getEmailForEnrolment" should {
+    "return the email address if found by CDS" in new TestCase {
+      private val connector = new ChannelPreferencesConnector(configuration, mockHttpClient)
+      when(
```

Comment:
shall we move mocks and test data to specialised TestContext/TestCase?
see example: https://github.com/hmrc/secure-message/blob/master/test/uk/gov/hmrc/securemessage/controllers/SecureMessageControllerSpec.scala#L56

Reply:
Done